contracts/Certificate.sol

```diff
@@ -133,8 +163,13 @@ contract Certificate is
   )
     internal
     override(ERC1155PresetMinterPauserUpgradeable, ERC1155SupplyUpgradeable)
+    whenNotPaused
   {
-    if (from != address(0)) {
+    bool isMinting = from != address(0);
```

Comment:
These two conditions look inverted to me.  `isMinting` is true when the from address _is_ `0`.  Likewise `isBurning` is true when to address _is_ `0`.